### ISSUE TITLE
Unshaded SnakeYAML classes packaged in the jar

### ISSUE THREAD

**Issue by papegaaij** [nielsbasjes/yauaa#961]
Since v7.14.1 (most likely the upgrade to snakeyaml 2.0), `yauaa-7.14.1.jar` contains 2 unshaded JVM 9 version classes from SnakeYAML:
```
$ jar tvf yauaa-7.14.1.jar | grep snakeyaml.internal.Logger 
  1678 Sun Feb 26 11:06:44 CET 2023 META-INF/versions/9/org/yaml/snakeyaml/internal/Logger$Level.class
  1484 Sun Feb 26 11:06:44 CET 2023 META-INF/versions/9/org/yaml/snakeyaml/internal/Logger.class
  1579 Sun Feb 26 11:06:42 CET 2023 nl/basjes/shaded/org/yaml/snakeyaml/internal/Logger$Level.class
  1383 Sun Feb 26 11:06:42 CET 2023 nl/basjes/shaded/org/yaml/snakeyaml/internal/Logger.class
```

These classes are causing errors in our build because the `maven-enforcer-plugin` forbids duplicate classes in multiple jars:
```
[ERROR] Rule 2: org.apache.maven.plugins.enforcer.BanDuplicateClasses failed with message:
[ERROR] Duplicate classes found:
[ERROR] 
[ERROR]   Found in:
[ERROR]     nl.basjes.parse.useragent:yauaa:jar:7.15.0:compile
[ERROR]     org.yaml:snakeyaml:jar:2.0:compile
[ERROR]   Duplicate classes:
[ERROR]     META-INF/versions/9/org/yaml/snakeyaml/internal/Logger.class
[ERROR]     META-INF/versions/9/org/yaml/snakeyaml/internal/Logger$Level.class
```

**Comment by nielsbasjes** [comment]
I assume your are using this one ?
https://www.mojohaus.org/extra-enforcer-rules/banDuplicateClasses.html

**Comment by nielsbasjes** [comment]
The root cause is indeed with snakeyaml itself:

```
$ unzip -l ~/.m2/repository/org/yaml/snakeyaml/1.33/snakeyaml-1.33.jar | fgrep Log
```
I.e. no Logger.class at all.

```
$ unzip -l ~/.m2/repository/org/yaml/snakeyaml/2.0/snakeyaml-2.0.jar | fgrep Log
     1545  2023-02-26 11:06   META-INF/versions/9/org/yaml/snakeyaml/internal/Logger$Level.class
     1365  2023-02-26 11:06   META-INF/versions/9/org/yaml/snakeyaml/internal/Logger.class
     1446  2023-02-26 11:06   org/yaml/snakeyaml/internal/Logger$Level.class
     1264  2023-02-26 11:06   org/yaml/snakeyaml/internal/Logger.class
```

**Comment by papegaaij** [comment]
Yes, that's the rule we are using for the enforcer. For now, we ignore these classes using ```                                    <dependency>
    <groupId>nl.basjes.parse.useragent</groupId>
    <artifactId>yauaa</artifactId>
    <ignoreClasses>
        <ignoreClass>META-INF.versions.9.org.yaml.snakeyaml.internal.*</ignoreClass>
    </ignoreClasses>
</dependency>
```

**Comment by nielsbasjes** [comment]
Just released v7.16.0 with a filter in the shading to drop these extra classes.
Please let me know if this works for you.

**Comment by papegaaij** [comment]
Yes, this fixes the issue. Thanks!
